src/Http/Exception/Client/MethodNotAllowedException.php

```diff
@@ -34,11 +35,35 @@ class MethodNotAllowedException extends ClientErrorException
     const DEFAULT_MESSAGE = 'A request was made using a HTTP method not supported by that resource';
 
     /**
+     * @var array
+     */
+    private $allowedMethods = [];
+
+    /**
      * @param string $message
      * @param mixed  $errors
+     * @param array  $allowedMethods
      */
-    public function __construct($message = '', $errors = null)
+    public function __construct($message = '', $errors = null, $allowedMethods = [])
     {
         parent::__construct(405, $message, $errors);
+        $this->allowedMethods = $allowedMethods;
+    }
+
+    /**
+     * Add the available methods (if any) to the Allow header
+     *
+     * {@inheritDoc}
+     */
+    public function prepareResponse(HttpResponse $response)
```

Comment:
I don't know, I did that long time ago and it's a nice and simple way to do that. I don't see why an exception that is explictly related to REST could not format the answer. That's the only place to do it without introducing too much complexity.